app/connectors/DESConnector.scala

```diff
@@ -17,28 +17,43 @@
 package connectors
 
 import config.BackendConfig
+import featureswitch.core.config.{FeatureSwitching, StubSubmission}
 import javax.inject.{Inject, Singleton}
-import models.submission.{DESSubmission, TopUpSubmission}
+import models.api.VatSubmission
+import models.submission.TopUpSubmission
 import play.api.Logger
-import play.api.libs.json.Writes
+import play.api.libs.json.{OWrites, Writes}
 import uk.gov.hmrc.http._
 import uk.gov.hmrc.http.logging.Authorization
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
-import scala.concurrent.ExecutionContext.Implicits.global
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class DESConnector @Inject()(val backendConfig: BackendConfig, val http: HttpClient) extends HttpErrorFunctions {
+class DESConnector @Inject()(val backendConfig: BackendConfig,
+                             val http: HttpClient
+                            )(implicit executionContext: ExecutionContext)
+  extends HttpErrorFunctions with FeatureSwitching {
 
-  lazy val desStubUrl: String = backendConfig.servicesConfig.baseUrl("des-stub")
-  lazy val desStubURI: String = backendConfig.servicesConfig.getConfString("des-stub.uri", "")
+  def des: String = if (isEnabled(StubSubmission)) {
+    s"$desStubUrl/$desStubURI"
+  }
+  else {
+    s"$desUrl/$desURI"
+  }
+
+  lazy val desStubUrl: String = backendConfig.servicesConfig.baseUrl("vat-registration")
+  lazy val desStubURI: String = backendConfig.servicesConfig.getConfString("des.stub.uri", "")
+  lazy val desUrl: String = backendConfig.servicesConfig.baseUrl("des.service")
+  lazy val desURI: String = backendConfig.servicesConfig.getConfString("des.service.uri", "")
```

Comment:
Do we need all of these? Would make it simpler to just have the url which uses baseUrl and adds on the extra in here

Reply:
Also these should be in BackendConfig

Reply:
Lines 38-48 essentially need refactoring and moving into BackendConfig (in the connector really all you should need to do is call backendConnector.desSubmissionUrl or something and it should figure out all the feature switching stuff in the config)